### PR TITLE
Add VEO overlay to dashboards

### PR DESCRIPTION
**Pull request by chnn**

### CODE REVIEW

File: chronograf/ui/mocks/dummy.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export const links: Links = {
   users: '/api/v2/users',
   write: '/api/v2/write',
   macros: '/api/v2/macros',
+  views: '/api/v2/views',
```
Comment (OfTheDelmer) [inline chronograf/ui/mocks/dummy.ts]: Nice

File: chronograf/ui/src/dashboards/actions/v2/hoverTime.ts [hunk 2]
```diff
@@ -1,17 +1,13 @@
-export enum ActionTypes {
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/actions/v2/hoverTime.ts]: Are we not continuing the `ActionTypes` pattern for actions?

File: chronograf/ui/src/dashboards/actions/v2/index.ts [hunk 3]
```diff
@@ -15,13 +14,15 @@ import {
   deleteCell as deleteCellAJAX,
   copyCell as copyCellAJAX,
 } from 'src/dashboards/apis/v2'
+import {createView as createViewAJAX} from 'src/dashboards/apis/v2/view'
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/actions/v2/index.ts]: Ahh, good to know this is here. I was going to start adding more view api calls to [`shared/apis/v2/view`](https://github.com/influxdata/platform/pull/999/files). Although, given that views are primarily used in dashboards I think this makes more sense.

File: chronograf/ui/src/dashboards/components/VEO.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  }
+}
+
+const mstp = (state: AppState): StateProps => {
+  const {viewName} = state.veo
+
+  return {viewName}
+}
+
+const mdtp: DispatchProps = {
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/components/VEO.tsx]: I like this separation of `DispatchProps`, `StateProps`,  and `PassedProps`.

File: chronograf/ui/src/dashboards/components/VEOHeaderName.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  }
+
+  public render() {
+    const {workingName, isEditing} = this.state
+
+    if (isEditing) {
+      return (
+        <div className="rename-dashboard">
+          <input
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/components/VEOHeaderName.tsx]: Does this need to be updated to the clockface input?

File: chronograf/ui/src/dashboards/components/VEOHeaderName.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            placeholder="Name this Cell..."
+            spellCheck={false}
+          />
+        </div>
+      )
+    }
+
+    return (
+      <div className="rename-dashboard">
+        <div className="rename-dashboard--title" onClick={this.handleClick}>
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/components/VEOHeaderName.tsx]: This also might be a clockface Button with `IconFont.pencil`

File: chronograf/ui/src/dashboards/components/VEOHeaderName.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  onRename: (name: string) => void
+}
+
+interface State {
+  workingName: string
+  isEditing: boolean
+}
+
+@ErrorHandling
+class CEOHeaderName extends Component<Props, State> {
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/components/VEOHeaderName.tsx]: Just needs to be updated to `VEOHeaderName` here and on export

File: chronograf/ui/src/dashboards/components/VEOHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      onSetViewName,
+      activeTab,
+      onSetActiveTab,
+      onCancel,
+      onSave,
+    } = this.props
+
+    return (
+      <div className="veo-header">
+        <Page.Header>
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/components/VEOHeader.tsx]: Nice!

File: chronograf/ui/src/dashboards/containers/DashboardPage.tsx [hunk 9]
```diff
@@ -41,63 +46,77 @@ import {
...
   links: Links
-  source: Source
   sources: Source[]
+  zoomedTimeRange: TimeRange
+  timeRange: TimeRange
+  dashboard: Dashboard
+  autoRefresh: number
+  inPresentationMode: boolean
+  showTemplateControlBar: boolean
+  views: {[viewID: string]: {view: View; status: RemoteDataState}}
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/containers/DashboardPage.tsx]: nice

File: chronograf/ui/src/dashboards/reducers/v2/views.ts [hunk 10]
```diff
@@ -1,16 +1,46 @@
...
-type State = string
+export interface ViewsState {
+  activeViewID: string
+  views: {
+    [viewID: string]: {
+      status: RemoteDataState
+      view: View
+    }
+  }
+  // viewStatus: RemoteDataState
```
Comment (OfTheDelmer) [inline chronograf/ui/src/dashboards/reducers/v2/views.ts]: Is this a stale comment?

File: chronograf/ui/src/dashboards/components/VEO.tsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+  onSetViewName: setViewName,
+}
+
+export default connect(mstp, mdtp)(VEO) as ComponentClass<PassedProps, State>
```
Comment (bthesorceror) [inline chronograf/ui/src/dashboards/components/VEO.tsx]: 👍

File: chronograf/ui/src/shared/components/cells/Cell.tsx [hunk 12]
```diff
@@ -1,56 +1,57 @@
...
-  }
-
+class CellComponent extends Component<Props> {
   public async componentDidMount() {
-    const {cell} = this.props
-    const view = await getView(cell.links.view)
-    this.setState({view, loading: RemoteDataState.Done})
+    const {viewStatus, cell, onReadView} = this.props
+
+    if (viewStatus === RemoteDataState.NotStarted) {
```
Comment (OfTheDelmer) [inline chronograf/ui/src/shared/components/cells/Cell.tsx]: Nice

File: chronograf/ui/src/shared/reducers/links.test.ts [hunk 13]
```diff
@@ -31,6 +31,7 @@ const links: Links = {
   users: '/api/v2/users',
   write: '/api/v2/write',
   macros: '/api/v2/macros',
+  views: '/api/v2/views',
```
Comment (OfTheDelmer) [inline chronograf/ui/src/shared/reducers/links.test.ts]: Nice

File: chronograf/ui/src/shared/reducers/links.ts [hunk 14]
```diff
@@ -24,6 +24,7 @@ const initialState: Links = {
   users: '',
   write: '',
   macros: '',
+  views: '',
```
Comment (OfTheDelmer) [inline chronograf/ui/src/shared/reducers/links.ts]: :)

File: chronograf/ui/src/dashboards/actions/v2/index.ts (hunk 3 above)
Comment (chnn) [reply chronograf/ui/src/dashboards/actions/v2/index.ts]: wups! should we keep these utils or the ones you have in #999?

File: chronograf/ui/src/dashboards/actions/v2/hoverTime.ts (hunk 2 above)
Comment (chnn) [reply chronograf/ui/src/dashboards/actions/v2/hoverTime.ts]: yeah i think we're just going to embed strings the the actual various action types. we achieve the same amount of type safety and don't have to write quite to much boilerplate

File: chronograf/ui/src/dashboards/components/VEOHeaderName.tsx (hunk 5 above)
Comment (chnn) [reply chronograf/ui/src/dashboards/components/VEOHeaderName.tsx]: prolly should, but it has some extra styling in addition. I'm going to punt on it for now (I think the real solution is to encapsulate this stuff in some sort of reusable rename component)

File: chronograf/ui/src/dashboards/components/VEOHeaderName.tsx (hunk 7 above)
Comment (chnn) [reply chronograf/ui/src/dashboards/components/VEOHeaderName.tsx]: nice catch

**Pull request merged**